### PR TITLE
[chore] move awscloudwatchlogs exporter to generated lifecycle tests

### PR DESCRIPTION
**Pull request by atoulme**
Relates to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27849

### CODE REVIEW

File: exporter/awscloudwatchlogsexporter/go.mod [hunk 1]
```diff
@@ -8,6 +8,7 @@ require (
 	github.com/google/uuid v1.5.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/awsutil v0.92.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/cwlogs v0.92.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.0.0-00010101000000-000000000000
```
Comment (songy23) [inline exporter/awscloudwatchlogsexporter/go.mod:11]: Wrong version?

**Review by bogdandrutu — APPROVED** [review]

File: exporter/awscloudwatchlogsexporter/go.mod (hunk 1 above)
Comment (atoulme) [reply exporter/awscloudwatchlogsexporter/go.mod:11]: it's replaced, will get fixed next release. I can fix it now.

Comment (atoulme) [reply exporter/awscloudwatchlogsexporter/go.mod:11]:
```suggestion
	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.92.0
```

**Review by bryan-aguilar — APPROVED** [review]

File: exporter/awscloudwatchlogsexporter/metadata.yaml [hunk 2]
```diff
@@ -7,3 +7,13 @@ status:
...
     active: [boostchicken, bryan-aguilar, rapphil]
+
+tests:
+  config:
+    endpoint: http://localhost:0
+    region: local
+    sending_queue:
+      enabled: false
+    retry_on_failure:
+      enabled: false
```
Comment (bryan-aguilar) [inline exporter/awscloudwatchlogsexporter/metadata.yaml:18]:
```suggestion
    log_group_name: "testing-logs"
    log_stream_name: "testing-integrations-stream"
```

Technically this is the smallest config required for the exporter

**Review by codeboten — APPROVED** [review]

**Pull request merged**